src/convert/from_10xh5_to_h5mu/script.py

```diff
@@ -1,32 +1,97 @@
-    "output": "5k_pbmc_protein_v3_nextgem_filtered_feature_bc_matrix.h5mu",
-}
-## VIASH END
-
+# set logging
 logger = logging.getLogger()
 logger.setLevel(logging.INFO)
 console_handler = logging.StreamHandler(stdout)
 logFormatter = logging.Formatter("%(asctime)s %(levelname)-8s %(message)s")
 console_handler.setFormatter(logFormatter)
 logger.addHandler(console_handler)
 
+## VIASH START
+par = {
+  "sample_id": "foo", 
+  "input": "resources_test/pbmc_1k_protein_v3/pbmc_1k_protein_v3_filtered_feature_bc_matrix.h5",
+  "input_metrics_summary": "resources_test/pbmc_1k_protein_v3/pbmc_1k_protein_v3_metrics_summary.csv",
+  "obs_sample_id": "sample_id",
+  "obsm_metrics": "metrics_summary",
+  "output": "foo.h5mu",
+  "id_to_obs_names": True,
+  "min_genes": 100,
+  "min_counts": 1000
+}
+## VIASH END
+
 logger.info("Reading %s.", par["input"])
-mdata = mu.read_10x_h5(par["input"])
+adata = sc.read_10x_h5(par["input"], gex_only=False)
+
+# store sample_id in .obs
+if par["sample_id"] is not None and par["obs_sample_id"] is not None:
+  logger.info(f"Storing sample_id '{par['sample_id']}' in .obs['{par['obs_sample_id']}]'.")
+  adata.obs[par["obs_sample_id"]] = par["sample_id"]
+
+# combine sample_id and barcode in obs_names
+if par["sample_id"] is not None and par["id_to_obs_names"] == True:
+  logger.info("Combining obs_names and sample_id")
+  replace = re.compile('-\\d+$')
+  adata.obs_names = [ replace.sub('', obs_name) + "_" + par["sample_id"] for obs_name in adata.obs_names ]
+
+# set the gene ids as var_names
+logger.info("Renaming var columns")
+adata.var = adata.var\
+  .rename_axis("gene_symbol")\
+  .reset_index()\
+  .set_index("gene_ids")
+
+# parse metrics summary file and store in .obsm or .obs
+if par["input_metrics_summary"] is not None:
+  logger.info(f"Reading metrics summary file '{par['input_metrics_summary']}'")
+  with open(par["input_metrics_summary"], newline='') as f:
+    reader = csv.reader(f)
+    data = list(reader)
+    header = data[0]
+    # header = [ s.lower().replace(' ', '_') for s in data[0] ]
+    # ^ do we want to remove spaces in the names?
+    
+    def string2floatorint(x):
+      y = re.sub("[^\\d\\.]", "", x)
+      if '%' in x:
+        return float(y) / 100
+      elif '.' in x:
+        return float(y)
+      else:
+        return int(y)
+    
+    values = [ string2floatorint(x) for x in data[1] ]
+    
+    metrics_summary = pd.DataFrame(dict(zip(header, values)), index=adata.obs_names)
+    
+  if par["obsm_metrics"] is not None:
+    logger.info(f"Storing metrics summary in .obs['{par['obsm_metrics']}']")
+    adata.obsm[par["obsm_metrics"]] = metrics_summary
+  else:
+    logger.info("Storing metrics summary in .obs")
+    adata.obs = adata.obs.join(metrics_summary)
+
+# might perform basic filtering to get rid of some data
+# applicable when starting from the raw counts
+if par["min_genes"] is not None:
+  logger.info(f"Filtering with min_genes={par['min_genes']}")
```

Comment:
Just a minor stylistic commnent: for logger the  recommended approach is not to use f-string formatting because the format module has functionality to format strings (using the old way of formatting) only when it is actually needed (which speeds up the script a bit).

```
logger.info("Filtering with min_genes=%s", par['min_genes'])
```
For multiple parameters to format you can use:
```
logger.info("Filtering with min_genes=%s. Another format: %s", par['min_genes'], 'foo')
```